agent/pkg/providers/stats_provider.go

```diff
@@ -3,26 +3,83 @@ package providers
 import (
 	"reflect"
 	"time"
+
+	"github.com/up9inc/mizu/tap/api"
 )
 
 type GeneralStats struct {
 	EntriesCount        int
 	EntriesVolumeInGB   float64
 	FirstEntryTimestamp int
 	LastEntryTimestamp  int
+	ProtocolCounters    map[string]map[string]*SizeAndRequestCount
 }
 
-var generalStats = GeneralStats{}
+type SizeAndRequestCount struct {
+	Count     int
+	BytesSize int
+}
+
+type AccumulativeStatsMethod struct {
+	MethodName   string `json:"methodName"`
+	RequestCount int    `json:"requestCount"`
+	ByteCount    int    `json:"byteCount"`
+}
+
+type AccumulativeStatsProtocol struct {
+	ProtocolName string                     `json:"protocolName"`
+	RequestCount int                        `json:"requestCount"`
+	ByteCount    int                        `json:"byteCount"`
+	Methods      []*AccumulativeStatsMethod `json:"methods"`
+}
+
+var generalStats = InitGeneralStats()
```

Comment:
There is a race around `generalStats`.
I guess that we mostly avoid it because `EntryAdded` is only called from one routine?
Still, there is no guarantee that this won't break.
Anyway there are `ResetGeneralStats` `GetAccumaltiveStats` which are called from different routines.
There should at least be a lock around `generalStats` (embed a Mutex in `GeneralStats` struct).